src/wcl/small_vector.h

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2022 SiFive, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You should have received a copy of LICENSE.Apache2 along with
+ * this software. If not, you may obtain a copy at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <algorithm>
+#include <vector>
+
+namespace wcl {
+
+// TODO: Use type traits to check if we can use realloc
+template <class T, size_t small_size>
+class alignas(T) small_vector {
```

Comment:
not sure if you need extra code for it or not, but this should probably be iterable right? that way it can be used in `foreach` loops